src/js/angular/getImgService.test.js

```diff
@@ -0,0 +1,134 @@
+                        secret: "1d4dc255d0",
+                        server: "693",
+                        title: ""
+                    }
+                ]
+            }
+
+
+    };
+
+        describe('should check the data taken by createdDomElm', function(){
+            var getImages = function(){
+                return ImgService.createImagesArray(sampleJSON);
+            };
+
+            it('should exist', function(){
+                expect(ImgService.createImagesArray).to.exist();
+            });
+            it('should return an array', function(){
+                expect(Array.isArray(getImages())).to.be.true();
+            });
+            it('should not be undefined', function(){
+                expect(getImages()).should.not.be.undefined();
+            });
+            it('should not be null', function(){
+                expect(typeof(getImages())).should.not.be.null();
+            });
+            it('should get the length of the object', function(){
+                expect(getImages().length).to.equal(2);
+            });
+        });
+
+        describe('should create an array with urls inside of it', function(){
+            it('should return something', function(){
+                expect(typeof(ImgService.createImagesArray(sampleJSON))).to.not.be.null();
+            });
+            it('should get each url as a string', function(){
+                var picsArray = ImgService.createImagesArray(sampleJSON);
+                for(var i = 0; i < picsArray.length; i++) {
+                    expect(typeof(picsArray[i])).to.equal('string');
+                }
+            });
+            it('should concatenate the contents of the given object', function() {
+                var createImagesArray = function (data) {
+                    var farm, server, id, title, secret, picUrl, currentPhoto;
+                    var images = [];
+
+                    for (var i = 0; i < data.photos.photo.length; i++) {
+                        currentPhoto = data.photos.photo[i];
+                        farm = currentPhoto.farm;
+                        server = currentPhoto.server;
+                        id = currentPhoto.id;
+                        title = currentPhoto.title;
+                        secret = currentPhoto.secret;
+                        picUrl = "https://farm" + farm + ".staticflickr.com/" + server + "/" + id + "_" + secret + "_z.jpg";
+                        images.push(picUrl);
+                        expect(images[i]).to.equal("https://farm" + farm + ".staticflickr.com/" + server + "/" + id + "_" + secret + "_z.jpg");
+                    }
+                };
+                createImagesArray(sampleJSON);
+            });
+        });
+
+    });
+});
+
+
+/*
+DIRECTIVE
+Accept a list of photos
+Generate URL for each photo
+Create a UL to house list
+Create a LI for each photo
+Each photo has click event
+** When photo clicked, open lightbox (check if lightbox class is applied)**
+ */
```

Comment:
Please remove these commented lines. Not needed in this service, but please take note if you want to use them for directive creation.